core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts

```diff
@@ -14,517 +14,611 @@
+        this.numberNonexistentIncompleteCollections = (
+          responseData.numberOfNonexistentActivities.incompleteCollections);
+        this.numberNonexistentCompletedExplorations = (
+          responseData.numberOfNonexistentActivities.completedExplorations);
+        this.numberNonexistentCompletedCollections = (
+          responseData.numberOfNonexistentActivities.completedCollections);
+        this.numberNonexistentExplorationsFromPlaylist = (
+          responseData.numberOfNonexistentActivities.explorationPlaylist);
+        this.numberNonexistentCollectionsFromPlaylist = (
+          responseData.numberOfNonexistentActivities.collectionPlaylist);
+        this.completedToIncompleteCollections = (
+          responseData.completedToIncompleteCollections);
+        this.threadSummaries = responseData.threadSummaries;
+        this.numberOfUnreadThreads =
+          responseData.numberOfUnreadThreads;
+        this.explorationPlaylist = responseData.explorationPlaylist;
+        this.collectionPlaylist = responseData.collectionPlaylist;
+        this.activeSection = (
+          LearnerDashboardPageConstants
+            .LEARNER_DASHBOARD_SECTION_I18N_IDS.INCOMPLETE);
+        this.activeSubsection = (
+          LearnerDashboardPageConstants
+            .LEARNER_DASHBOARD_SUBSECTION_I18N_IDS.EXPLORATIONS);
+        this.feedbackThreadActive = false;
+
+        this.noExplorationActivity = (
+          (this.completedExplorationsList.length === 0) &&
+            (this.incompleteExplorationsList.length === 0));
+        this.noCollectionActivity = (
+          (this.completedCollectionsList.length === 0) &&
+            (this.incompleteCollectionsList.length === 0));
+        this.noActivity = (
+          (this.noExplorationActivity) && (this.noCollectionActivity) &&
+          (this.explorationPlaylist.length === 0) &&
+          (this.collectionPlaylist.length === 0));
+      }, errorResponseStatus => {
+        if (
+          AppConstants.FATAL_ERROR_CODES.indexOf(errorResponseStatus) !== -1) {
+          this.alertsService.addWarning(
+            'Failed to get learner dashboard data');
         }
-      };
-
-      ctrl.setSubscriptionSortingOptions = function(sortType) {
-        if (sortType === ctrl.currentSubscribersSortType) {
-          ctrl.isCurrentSubscriptionSortDescending = (
-            !ctrl.isCurrentSubscriptionSortDescending);
-        } else {
-          ctrl.currentSubscribersSortType = sortType;
+      }
+    );
+
+
+    Promise.all([
+      userInfoPromise,
+      dashboardDataPromise,
+    ]).then(() => {
+      // The timeout is required because at execution time,
+      // the element may not be present in the DOM yet.Thus it ensure
+      // that the element is visible before focussing.
+      setTimeout(() => {
+        this.loaderService.hideLoadingScreen();
+        this.addFocusWithoutScroll('ourLessonsBtn');
+      }, 0);
+    // eslint-disable-next-line dot-notation
```

Comment:
Why is this needed?

Reply:
.delete is throwing lint error,
(Other parts of the codebase also uses this.)